src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift

```diff
@@ -20,48 +20,87 @@
 import Foundation
 
 final class CachedAppConfiguration {
-	// MARK: Creating a Cached App Configuration
-	init(client: Client) {
+
+	enum CacheError: Error {
+		case dataFetchError(message: String?)
+		case dataVerificationError(message: String?)
+		/// HTTP 304 – Content on server has not changed from the given `If-None-Match` header in the request
+		case notModified
+	}
+
+	/// Most likely a HTTP client
+	private let client: AppConfigurationFetching
+
+	/// The place where the app config and last etag is stored
+	private let store: AppConfigCaching
+
+	init(client: AppConfigurationFetching, store: AppConfigCaching) {
 		self.client = client
+		self.store = store
+
+		guard shouldFetch() else { return }
+
+		// edge case: if no app config is cached, omit a potentially existing ETag to force fetch a new configuration
+		let etag = store.appConfig == nil ? nil : store.lastAppConfigETag
```

Comment:
Can lastAppConfigETag exist without  a config?

Reply:
We currently have a very naïve store which does not provide any validation. We can in fact set/store ETags without app configuration. Future refactorings of the Store could clear the ETag if a config is deleted but that should be refactored later and not as part of this PR.

Reply:
Maybe a solution is lastAppConfigETag and appConfig should be combined in a struct?

Reply:
In the database etag, config and further cache control is stored in separate fields. We need a setter for the app config that checks and verifies the pre- and postconditions on changes. I think, a simple struct doesn't help.

Reply:
Ok, good point.